backend/controllers/LocationsController.ts

```diff
@@ -91,6 +92,27 @@ const LocationsController = {
       return res.status(500).json({ error: err.message });
     }
   },
-};
+
+  async addLocationComments(req: Request, res: Response) {
+    try {
+      const {id,comment}=req.body;
+      const locations = await Location.findById(id);
+      if (!locations||!locations.comments) {
+        return res.status(400).json({ error: "Location or comments doesn't exist" });
+      }
```

Comment:
findById returns a single object, therefore you should call the variable "location", not "locationS"